src/kv/kv.h

```diff
@@ -1670,7 +1698,9 @@ namespace kv
         version,
         (globally_committable ? " globally_committable" : ""));
 
-      std::vector<std::tuple<Version, std::vector<uint8_t>, bool>> batch;
+      std::vector<
+        std::tuple<Version, std::shared_ptr<flatbuffers::DetachedBuffer>, bool>>
```

Comment:
do you need the thread safety here? If not can this just be wrapped in a different smart pointer, even it is not one from the stl?

Reply:
As discussed offline, replaced shared pointer with unique pointer. `std::function`s need to be copy-able, so created an intermediate class `MovePendingTx` which acts as our functor. The class also provides the copy constructor and copy assignment operator, but throws an error if they are actually called. We make sure this way that the lambda passed into `store->commit` is being moved, and as a result the objects that it captures are also being moved.